### PR TITLE
Add F1, precision, recall to metrics package

### PR DESCRIPTION
**Pull request by ngreenwald**
Currently the metrics package doesn't include some of the default parameters we care about, and they have to be calculated separately. This PR adds them in. It also removes Python 2.7 from testing since it is no longer supported.

### CODE REVIEW

**Comment by ngreenwald** [comment]
Tests are failing due to different behavior in 2.x. Now that we've dropped 2.x from deepcell-tf, do we need to keep it here?

**Comment by willgraf** [comment]
Yes, it's no problem to move on from Python 2, but in this case, it is not Python 2 that is different, but a different version of `scipy`. Please find the minimum version required for `scipy` in this PR and update the requirements/setup.py to reflect that floor.

**Review by willgraf — CHANGES_REQUESTED** [review]
Looks fine but let's update the `scipy` version requirement in `setup.py` to reflect the breaking change in `hmean`. (Or find a substitute function that does not have breaking changes).

**Comment by ngreenwald** [comment]
Updated the requirements. There's not a matching distribution that works for 2.X

**Comment by willgraf** [comment]
Please also update the GitHub workflows to not test 2.7, and make it clear in the PR that this PR removes support for Python 2.7.

**Comment by ngreenwald** [comment]
Oh nice, I didn't realize that's where the github settings were configured

File: deepcell_toolbox/metrics.py [hunk 1]
```diff
@@ -598,6 +604,11 @@ def save_to_dataframe(self):
         Returns:
             pandas.DataFrame: Single row dataframe with error types as columns
         """
+
+        self.precision = self.correct_detections / self.n_pred
+        self.recall = self.correct_detections / self.n_true
+        self.f1 = hmean([self.recall, self.precision])
```
Comment (willgraf) [inline deepcell_toolbox/metrics.py:610]: Do we really need `hmean`?

Could we just do:

```python
self.f1 = 2*(self.recall * self.precision) / (self.recall + self.precision)
```

File: deepcell_toolbox/metrics.py [hunk 2]
```diff
@@ -57,6 +57,8 @@
 from skimage.measure import regionprops
 from skimage.segmentation import relabel_sequential
 from sklearn.metrics import confusion_matrix
+from scipy.stats import hmean
+
```
Comment (willgraf) [inline deepcell_toolbox/metrics.py:61]: Remove the unnecessary whitespace

```suggestion
```

File: deepcell_toolbox/metrics.py (hunk 1 above)
Comment (ngreenwald) [reply deepcell_toolbox/metrics.py:610]: Yeah, the reason I was using hmean was because Dave used to compute F1 back when we had to do it on the output for the Metrics package. Now that we're doing it internally on each loop, we only have two values, rather than a list of like 300 values each. I'll swap it.

Comment (willgraf) [reply deepcell_toolbox/metrics.py:610]: Reflecting on it, I see no need to swap it out. I'm happy with the `hmean` build.

**Review by willgraf — APPROVED** [review]
LGTM.
